### ISSUE TITLE
slimerjs: line 173: 252476 Segmentation fault      "$SLIMERJSLAUNCHER" -app "$SLIMERDIR/application.ini" $PROFILE -no-remote "$@" 2> /dev/null

### ISSUE THREAD

**Issue by mingdaolu** [laurentj/slimerjs#338]
I use caperjs + slimerjs to test my web page.

but when i run the test cases,  i got following error randomly:
0.9.5/bin/slimerjs: line 173: 252476 Segmentation fault      "$SLIMERJSLAUNCHER" -app "$SLIMERDIR/application.ini" $PROFILE -no-remote "$@" 2> /dev/null
 
The version i used is:
casperjs 1.1-beta3
slimerjs 0.9.5

Due to this issue pop up randomly,  sometimes it always work, other times the error occurs every time.

**Comment by ludovicPelle** [comment]
+1

**Comment by mingdaolu** [comment]
Do you need any more information to investigate this issue?

**Comment by 4e976623** [comment]
I think that http://livejournal.com/usmers/_niece/ reliably crashes SlimerJS snapshot as of 2015-08-20 with Firefox 40.0.2 used as XULRunner. The crash occurs every single time I try. The same Firefox opens the same URL fine. Other livejournal,com blogs seem to work fine. I start with an empty profile without livejournal.com cookies and don't attempt to log in. 

I think this issue is general enough so my example is not derailing or hijacking it.

Apparently traceback includes NS_InvokeByIndex…

My system is x86_64 GNU/Linux with X11 (Firefox is from Nixpkgs, but I _really_ hope that doesn't matter; I tried both our form-source build and Firefox binary release). Can anyone confirm this on another distribution? Or maybe on i686 or MacOS X?

**Comment by laurentj** [comment]
@4e976623 what is the error message ?

every informations may help, as I cannot reproduce the issue (on linux 64b).

**Comment by 4e976623** [comment]
>@4e976623 what is the error message ?
>
>every informations may help, as I cannot reproduce the issue (on linux 64b).

Console at 330 in http://l-stat.livejournal.net/js/??.ljlib.js?v=1440415814: Widget text variable [collapse] hasn't been defined
Console at 330 in http://l-stat.livejournal.net/js/??.ljlib.js?v=1440415814: Widget text variable [collapse] hasn't been defined
Console at 330 in http://l-stat.livejournal.net/js/??.ljlib.js?v=1440415814: Widget text variable [collapse] hasn't been defined
/nix/store/kzmjfyy4ahf2v48nzyly014xfxyad8cp-slimerjs-0.9.6.2015.08.20/lib/slimerjs/slimerjs: line 179:  6662 Segmentation fault      "$SLIMERJSLAUNCHER" -app "$SLIMERDIR/application.ini" $PROFILE -no-remote "$@" 2> /dev/null

This is the error message I get. Not too informative, unfortunately.

I was going to try getting Ubuntu LiveCD, getting Firefox binary 40.0.2 
and running the same slimer version, but I haven't yet got around to it,
so I send what I already have and will give more information as I check.

Thanks for the reproduction, the fact that this site doesn't crash this
revision of SlimerJS everywhere brings some optimism. Maybe there are 
some subtle library version requirements…

**Comment by SwCharlie** [comment]
Temporary fix:
export SLIMERJSLAUNCHER=/usr/bin/firefox in CLI.
Permanent fix:
in /usr/local/lib/node_modules/slimerjs/lib/slimer/slimerjs (or where-ever slimerjs is)
replace:
SLIMERJSLAUNCHER="$SLIMERDIR/xulrunner/xulrunner"
with:
#SLIMERJSLAUNCHER="$SLIMERDIR/xulrunner/xulrunner"
SLIMERJSLAUNCHER="/usr/bin/firefox"

**Comment by 4e976623** [comment]
Doesn't remove the crash here.

**Comment by SwCharlie** [comment]
using Mozilla Firefox 38.2.1. You are using 40? Ensure your firefox is also in the location /usr/bin/firefox or where ever you specified it (command: which firefox).

**Comment by 4e976623** [comment]
>using Mozilla Firefox 38.2.1. You are using 40? Ensure your firefox is also in the location /usr/bin/firefox or where ever you specified it (command: which firefox).

Yes, I think all was fine back with 38.2.1. Now I have 40.0.3 here.

**Comment by SwCharlie** [comment]
I think only up to 38 or 39 is supported at this stage.

**Comment by laurentj** [comment]
have you got a minimal test case showing the issue? + slimerjs version + firefox version + operating system.

**Comment by 4e976623** [comment]
After some updates (maybe local, maybe site updates) the sites that 
reliably crashed SlimerJS personally for me do not crash it now. I don't
know about others.

**Comment by SwCharlie** [comment]
Some reason this error seems to happen when trying to install Firefox any other method other than going: yum install firefox. This limits me to Mozilla Firefox 38.3.0, and I wish to upgrade/downgrade :(

**Comment by wtomed** [comment]
In my case it was an incomplete configuration of dbus that caused the problem. The seg fault message comes from xvfb-run, and is not the problem. To test if it's dbus config causing the problem, run: xvfb-run /usr/bin/firefox. Firefox will exit with instructions on how to fix.
